### PR TITLE
Tests for the model

### PR DESCRIPTION
**Pull request by DelaneySteve**

### CODE REVIEW

File: data/unit_tests/make_mock_dataset.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import json
```
Comment (james-zafar) [inline data/unit_tests/make_mock_dataset.py]: I'm not sure I understand the rationale for this - if we have data already gathered from fr24 and other sources, would it not be better to use a subset of that?

File: data/unit_tests/make_mock_dataset.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+MAX_NUMBER_FLIGHTS = 500
+MAX_NUMBER_RUNWAYS = 16
+MAX_NUMBER_AIRPORTS = 100
+
+
+def fake_airports_generator(length: int) -> list[dict[str, Any]]:
+    airports_list = []
+    for _ in range(length):  # xrange in Python 2.7
+        og_iata = fake.airport_iata()
+        og_icao = fake.airport_icao()
```
Comment (james-zafar) [inline data/unit_tests/make_mock_dataset.py]: How do we ensure the `icao` and `iata` are consistent with one another?

File: data/unit_tests/make_mock_dataset.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        airports_list.append(airport)
+    return airports_list
+
+
+def fake_flight_list(length: int, og_iata: str, og_icao: str) -> list[dict[str, Any]]:
+    flight_list = []
+    for _ in range(length):  # xrange in Python 2.7
+        if randint(0, 10) > 3:  # set to og_iata
+            flight = {
+                'aircraft_model': uniform(0, 100),
```
Comment (james-zafar) [inline data/unit_tests/make_mock_dataset.py]: Is an integer a valid aircraft model?

File: data/unit_tests/make_mock_dataset.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                'scheduled_arrival': randint(1700000000, 1800000000),
+                'scheduled_departure': randint(1700000000, 1800000000)
+            }
+        flight_list.append(flight)
+    return flight_list
+
+
+def fake_runway_list(length: int) -> list[dict[str, Any]]:
+    runway_list = []
+    for _ in range(length):  # xrange in Python 2.7
```
Comment (james-zafar) [inline data/unit_tests/make_mock_dataset.py]: Is this code copied from somewhere? The `xrange in Python 2.7` comment is unnecessary

File: src/model/model.py [hunk 5]
```diff
@@ -30,7 +31,7 @@ def load_trained_model(cls, filename: str) -> 'Model':
         instance._model = model
         return instance
 
-    def predict(self, prepped_input: pd.DataFrame) -> float:
+    def predict(self, prepped_input: pd.DataFrame) -> Any:
```
Comment (james-zafar) [inline src/model/model.py]: Why has this changed to `Any`?

File: tests/model/test_model.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        'total_runway_length',
+        'total_arrivals',
+        'total_departures',
+    ]
+    TARGET = ['air_quality']
+
+    def setUp(self) -> None:
+        self.model = Model()
+        self.data_path = 'data/unit_tests/model/mock_airport_dataset.json'
+        self.model_path = 'src/model/model_new.pickle'
```
Comment (james-zafar) [inline tests/model/test_model.py]: Perhaps we could name this something more sensible to make it clear it is to be used for tests e.g. `random_forest_test_model` or something?

File: tests/model/test_model.py [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+        with open(self.data_path, 'r', encoding='utf-8') as f:
+            self.raw_data = json.load(f)
+
+    def test_loading_wrong_model(self) -> None:
+        wrong_model_path = 'data/unit_tests/model/lr_model.pickle'
```
Comment (james-zafar) [inline tests/model/test_model.py]: What is inside this model?

File: tests/model/test_model.py [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+            test_model_class.load_trained_model(wrong_model_path)
+
+    def test_data_preprocessing(self) -> None:
+        targets, features = self.model.preprocessing(self.raw_data)
+        self.assertIsInstance(features, pd.DataFrame)
+        self.assertIsInstance(targets, pd.DataFrame)
+        self.assertEqual(features.shape[1], len(self.FEATURE_COLUMNS))
+        self.assertEqual(targets.shape[1], len(self.TARGET))
+        self.assertEqual(set(features.columns), set(self.FEATURE_COLUMNS))
+        self.assertEqual(set(targets.columns), set(self.TARGET))
```
Comment (james-zafar) [inline tests/model/test_model.py:43]: Should we not check that the actual data is correct also?

File: data/unit_tests/make_mock_dataset.py (hunk 2 above)
Comment (DelaneySteve) [reply data/unit_tests/make_mock_dataset.py:25]: icao is just a placeholder we dont use it :/

File: tests/model/test_model.py (hunk 7 above)
Comment (DelaneySteve) [reply tests/model/test_model.py]: a basic linear regression model

File: data/unit_tests/make_mock_dataset.py (hunk 1 above)
Comment (DelaneySteve) [reply data/unit_tests/make_mock_dataset.py]: Oh is using a subset okay? I thought we didn't want to upload the dataset to the repo?

File: src/api/operations/create_prediction.py [hunk 9]
```diff
@@ -21,7 +21,7 @@
 API_KEY = os.getenv('API_KEY')
 api_key_header = APIKeyHeader(name='X-API-key')
 
-MODEL_PATH: Final[str] = './model/rf_model.pickle'
+MODEL_PATH: Final[str] = './data/models/rf_model.pickle'
```
Comment (james-zafar) [inline src/api/operations/create_prediction.py]: The production model should still be in `./model/...`

**Review by james-zafar — APPROVED** [review]
Looks good, thanks

**Pull request merged**